packages/workbox-precaching/src/PrecacheController.ts

```diff
@@ -62,13 +63,23 @@ class PrecacheController {
 
     for (const entry of entries) {
       const {cacheKey, url} = createCacheKey(entry);
+
       if (this._urlsToCacheKeys.has(url) &&
           this._urlsToCacheKeys.get(url) !== cacheKey) {
         throw new WorkboxError('add-to-cache-list-conflicting-entries', {
           firstEntry: this._urlsToCacheKeys.get(url),
           secondEntry: cacheKey,
         });
       }
+
+      if (typeof entry !== 'string' && entry.integrity) {
```

Comment:
I'm not 100% sure that this is the most TypeScript-y way of performing this conditional logic.

Reply:
I think this is fine. Does TypeScript complain if you just check `entry.integrity`?

Reply:
Yeah, without the first condition I end up with:

```
Property 'integrity' does not exist on type 'string | PrecacheEntry'.
  Property 'integrity' does not exist on type 'string'.ts(2339)
```